data/transactions/logic/eval.go

```diff
@@ -2418,7 +2426,7 @@ func (cx *EvalContext) opTxnImpl(gi uint64, src txnSource, field TxnField, ai ui
 	tx := &group[gi]
 
 	// int(gi) is safe because gi < len(group). Slices in Go cannot exceed `int`
-	sv, err = cx.txnFieldToStack(tx, fs, ai, int(gi), src != srcGroup)
+	sv, err = cx.txnFieldToStack(tx, fs, ai, startIdx+int(gi), src != srcGroup)
```

Comment:
I don't think we can do this.  If we lie about the `gi`, it'll mess up `gtxn 4 GroupIndex` (though I admit that is a ridiculous thing to do, since it has to be 4!).  We probably need to calculate the offset and pass it in separately.